reference/reference.md

```diff
@@ -0,0 +1,273 @@
+# Reference documentation
+> Note: This section of the ngrx-auto-entity documentation is a work in progress. A lot of features are currently missing, incomplete, or out of date. Feel free to ask questions on github or by e-mail if needed!
+
+<!-- -->
+<br>
+<br>
+<!-- -->
+
+<!-- Section: Modules -->
+## **Modules**
+
+<!-- NgrxAutoEntityModule -->
+`NgrxAutoEntityModule` - The main module that should be imported into the app module or feature modules.
+
+**For the AppModule:**
+{% code-tabs %}
+{% code-tabs-item title="src/app/app.module.ts" %}
+```typescript
+import { NgrxAutoEntityModule } from '@briebug/ngrx-auto-entity';
+// ... other imports ...
+
+@NgModule({
+    imports: [
+        BrowserModule,
+        StoreModule.forRoot(appReducer, { metaReducers: appMetaReducers }),
+        EffectsModule.forRoot([]),
+        NgrxAutoEntityModule.forRoot() // Add this!
+    ]
+})
+export class AppModule {}
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+**For a feature module:**
+{% code-tabs %}
+{% code-tabs-item title="src/app/+orders/orders.module.ts" %}
+```typescript
+import { NgrxAutoEntityModule } from '@briebug/ngrx-auto-entity';
+// ... other imports ...
+
+@NgModule({
+    imports: [
+        CommonModule,
+        StoreModule.forFeature('orders', featureReducer),
+        NgrxAutoEntityModule.forFeature() // Changed from forRoot() to forFeature()
+    ]
+})
+export class AppModule {}
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+<!-- -->
+<br>
+<br>
+<!-- -->
+
+<!-- Section: Functions -->
+## **Functions**
+
+<!-- buildState -->
+`buildState` - Builds the initial NgRx state for an entity and provides a base Facade class that can be extended.
+
+```typescript
+buildState(type: IModelClass<TModel>, extraInitialState?: any): IModelState<TParentState, TState, TModel>
+```
+
+| Parameter | Type | Description |
+| --------- | ---- | ----------- |
+| type | IModelClass\<TModel\> | the entity class |
+| *extraInitialState?* | *any* | *the (optional) initial state* |
+
+**Returns** [IModelState](#imodelstate)
+
+<br>
+<br>
+
+<!-- buildFeatureState -->
+`buildFeatureState` - Builds the Ngrx state for an entity that is part of a feature module.
+
+```typescript
+buildFeatureState<TState extends IEntityState<TModel>, TParentState, TModel>(
+  type: IModelClass<TModel>,
+  featureStateName: NonNullable<string>,
+  selectParentState: MemoizedSelector<object, TParentState>,
+  extraInitialState?: any
+): IModelState<TParentState, TState, TModel>
+```
+
+| Parameter | Type | Description |
+| --------- | ---- | ----------- |
+| type | IModelClass\<TModel\> | the entity class |
+| featureStateName | NonNullable\<string\> | the name of the feature state |
+| selectParentState | MemoizedSelector\<object, TParentState\> | a selector for the entity's parent state |
+| *extraInitialState?* | *any* | *the (optional) initial feature state* |

```

Comment:
This can be initial state for state Auto-Entity manages, as well as any additional state the developer wishes to include for this entity.

Reply:
Added new commit with changes, + an old change I forgot to commit a while back. Take a look when you have time!